File: Code/app/src/main/java/com/bigfriedicecream/recipes/views/RecipeListFragment.java
package com.bigfriedicecream.recipes.views;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.bigfriedicecream.recipes.R;
import com.bigfriedicecream.recipes.interfaces.IRecipeListContract;
import com.bigfriedicecream.recipes.models.RecipeDataModel;
import com.bigfriedicecream.recipes.presenters.RecipeListPresenter;
import com.google.gson.Gson;

import java.util.Map;

public class RecipeListFragment extends Fragment implements IRecipeListContract.View {

    private IRecipeListContract.Presenter presenter;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        presenter = new RecipeListPresenter(this);
        return inflater.inflate(R.layout.fragment_recipe_list, container, false);
    }

    @Override
    public void onStart() {
        super.onStart();
        presenter.start(getContext());
    }

    @Override
    public void onStop() {
        super.onStop();
        presenter.stop();
    }

    public void render(Map<String, RecipeDataModel> recipeList) {
        LinearLayout layoutList = getView().findViewById(R.id.layout_list);
        layoutList.removeAllViews();

        for (Map.Entry<String, RecipeDataModel> entry : recipeList.entrySet()) {
            Bundle bundle = new Bundle();
            bundle.putString("data", new Gson().toJson(entry.getValue()));

            Fragment fragment = new RecipeListItemFragment();
            fragment.setArguments(bundle);

            getChildFragmentManager()
                    .beginTransaction()
                    .add(R.id.layout_list, fragment)
                    .commit();
        }
    }
}

File: Code/app/src/main/java/com/twobrothers/overcooked/views/recipelist/RecipeListFragment.kt
package com.twobrothers.overcooked.views.recipelist

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.twobrothers.overcooked.R
import com.twobrothers.overcooked.interfaces.IRecipeListContract
import com.twobrothers.overcooked.presenters.RecipeListPresenter
import com.twobrothers.overcooked.utils.RecipeListViewAdapter

class RecipeListFragment : Fragment(), IRecipeListContract.View {

    private val presenter: IRecipeListContract.Presenter = RecipeListPresenter(this)

    private lateinit var recyclerView: RecyclerView
    private lateinit var viewAdapter: RecyclerView.Adapter<*>
    private lateinit var viewManager: RecyclerView.LayoutManager

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_recipe_list, container, false)

        viewManager = LinearLayoutManager(context)
        viewAdapter = RecipeListViewAdapter(presenter, mutableListOf())

        recyclerView = view!!.findViewById<RecyclerView>(R.id.recycler_container).apply {
            layoutManager = viewManager
            adapter = viewAdapter
        }

        return view
    }

    override fun onStart() {
        super.onStart()
        presenter.onStart()
    }

    override fun onStop() {
        super.onStop()
        presenter.onStop()
    }

    override fun render() {
        viewAdapter.notifyDataSetChanged()
    }
}
